### PR TITLE
wifi: AP connection support

### PR DESCRIPTION
**Pull request by krish2718**

### CODE REVIEW

File: drivers/wifi/nrf700x/src/wpa_supp_if.c [hunk 1]
```diff
@@ -1819,6 +1819,32 @@ static int nrf_wifi_iftype_change(struct nrf_wifi_vif_ctx_zep *vif_ctx_zep, int
 	return ret;
 }
 
+static int nrf_wifi_wait_for_carrier_status(struct nrf_wifi_vif_ctx_zep *vif_ctx_zep,
```
Comment (markaj-nordic) [inline drivers/wifi/nrf700x/src/wpa_supp_if.c:1822]: I see that fmac_main.c AP related parts are guarded with `CONFIG_NRF700X_AP_MODE`. Shouldn't we do the same with the wpa_supp bits as well?

Comment (krish2718) [reply drivers/wifi/nrf700x/src/wpa_supp_if.c:1822]: It is guarded, see line#1728.

File: drivers/wifi/nrf700x/src/wpa_supp_if.c [hunk 2]
```diff
@@ -2060,16 +2175,224 @@ int nrf_wifi_wpa_supp_deinit_ap(void *if_priv)
...
+		LOG_ERR("%s: nrf_wifi_fmac_add_sta failed", __func__);
+		goto out;
+	}
+
+	ret = 0;
+out:
+	return ret;
+}
+
+/* TODO: Move this to UMAC headers */
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/src/wpa_supp_if.c:2271]: Can we take care of this before merging this PR ?

**Review by sachinthegreen — APPROVED** [review]

**Comment by krish2718** [comment]
Zephyr manifest update is superceded by https://github.com/nrfconnect/sdk-nrf/pull/13466, so, will rebase once its merged.

**Review by rado17 — APPROVED** [review]

**Pull request merged**